ydb/docs/en/core/reference/observability/tracing/setup.md

```diff
@@ -0,0 +1,348 @@
+|  0 | `TopLevel` | `Off` | `Off` | `Off` | `Off` |
+|  1 | `TopLevel` | **`TopLevel`** | `Off` | `Off` | `Off` |
+|  2 | `TopLevel` | `TopLevel` | **`TopLevel`** | `Off` | `Off` |
+|  3 | `TopLevel` | `TopLevel` | `TopLevel` | **`TopLevel`** | `Off` |
+|  4 | `TopLevel` | `TopLevel` | `TopLevel` | `TopLevel` | **`TopLevel`** |
+|  5 | **`Basic`** | `TopLevel` | `TopLevel` | `TopLevel` | `TopLevel` |
+|  6 | `Basic` | **`Basic`** | `TopLevel` | `TopLevel` | `TopLevel` |
+|  7 | `Basic` | `Basic` | **`Basic`** | `TopLevel` | `TopLevel` |
+|  8 | `Basic` | `Basic` | `Basic` | **`Basic`** | `TopLevel` |
+|  9 | `Basic` | `Basic` | `Basic` | `Basic` | **`Basic`** |
+| 10 | **`Detailed`** | **`Detailed`** | `Basic` | `Basic` | `Basic` |
+| 11 | `Detailed` | `Detailed` | **`Detailed`** | `Basic` | `Basic` |
+| 12 | `Detailed` | `Detailed` | `Detailed` | **`Detailed`** | `Basic` |
+| 13 | `Detailed` | `Detailed` | `Detailed` | `Detailed` | **`Detailed`** |
+| 14 | **`Diagnostic`** | **`Diagnostic`** | **`Diagnostic`** | **`Diagnostic`** | **`Diagnostic`** |
+| 15 | **`Trace`** | **`Trace`** | **`Trace`** | **`Trace`** | **`Trace`** |
+
+### Rules
+
+#### External Throttling
+
+The semantics of each rule are such that it allocates a quota for the number of requests of a given category. For example, if the `external_throttling` section looks like this:
+
+```yaml
+tracing_config:
+  external_throttling:
+    - max_traces_per_minute: 60
+    - scope:
+        request_types:
+          - KeyValue.ReadRange
+      max_traces_per_minute: 20
+```
+
+With a sufficient flow of requests with an external trace-id, at least 60 requests per minute and at least 20 requests of type `KeyValue.ReadRange` will be traced per minute. In total, no more than 80 requests per minute will be traced.
+
+The decision algorithm is as follows: for a request with an external trace-id, the set of rules under which the request falls is determined. The request consumes the quota from all rules where it is still available. A request is not traced only if no rule has any quota left.
+
+#### Sampling
+
+The semantics of the sampling rule are similar: with a sufficiently small flow of requests of a given category, at least `fraction` of the requests will be sampled with at least `level` verbosity level.
+
+The decision algorithm is similar: for a request without an external trace-id (either due to its initial absence or due to a previous decision not to trace this request), the set of rules under which the request falls is determined. The request consumes the quota from all rules where it is still available and which randomly "decided" to sample it. If the request does not need to be sampled according to any rule (in all rules that randomly "decided" to sample the request, no quota is left), it is not sampled. Otherwise, the verbosity level is determined as the maximum among the rules into which the request fell.
```

Comment:
```suggestion
The decision algorithm is similar: the set of rules under which the request falls is determined for a request without an external `trace-id` either due to its initial absence or due to a previous decision not to trace this request. The request consumes the quota from all rules where it is still available and which randomly "decided" to sample it. It is not sampled if the request does not need to be sampled according to any rule (in all rules that randomly "decided" to sample the request, no quota is left). Otherwise, the verbosity level is determined as the maximum among the rules into which the request fell.
```